packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -43,16 +42,11 @@ import {getAggregationBits, getAggregatorModulo} from "./utils";
 export class AttestationService {
   private readonly config: IBeaconConfig;
   private readonly provider: IApiClient;
-  // order is important
-  private readonly secretKeys: SecretKey[] = [];
-  // order is important
-  private readonly publicKeys: BLSPubkey[] = [];
-  // order is important
-  private readonly validators: (ValidatorResponse | null)[] = [];
+  private readonly validators: Map<string, ValidatorAndSecret> = new Map();
   private readonly slashingProtection: ISlashingProtection;
   private readonly logger: ILogger;
 
-  private nextAttesterDuties: Map<Slot, Map<number, IAttesterDuty>> = new Map();
+  private nextAttesterDuties: Map<Slot, Map<BLSPubkey, IAttesterDuty>> = new Map();
```

Comment:
is there a reason to use `BLSPubkey` as a key for Map? Normally we use number or string as key for Map to easily compare by value, it should help with `Map.get()`, `Map.has()` etc.

Reply:
nope, no good reason. i meant to change this in accordance with the new `validators` object (which uses the string instead of BLSPubKey).  will change this